Source/AliveLibAO/Abe.cpp

```diff
@@ -2738,11 +2740,393 @@ void Abe::VOn_Tlv_Collision_421130(Path_TLV *pTlv)
+    {
+        if (sInputObject_5009E8.isHeld(sInputKey_ThrowItem_4C65B4) &&
+            field_FC_current_motion == eAbeStates::State_0_Idle_423520)
+        {
+            if (field_19C_throwable_count > 0 || gInfiniteGrenades_5076EC)
+            {
+                field_198_pThrowable = Make_Throwable_454560(
+                    field_A8_xpos,
+                    field_AC_ypos - FP_FromInteger(40),
+                    0
+                );
+
+                if (bThrowableIndicatorExists_504C70 == 0)
+                {
+                    auto pThrowableTotalIndicator = ao_new<ThrowableTotalIndicator>();
+                    if (pThrowableTotalIndicator)
+                    {
+                        const FP xOffSet = field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX) ?
+                            FP_FromInteger(15) : FP_FromInteger(-15) * field_BC_sprite_scale;
+
+                        pThrowableTotalIndicator->ctor_41B520(
+                            field_A8_xpos + xOffSet,
+                            field_AC_ypos + field_BC_sprite_scale * FP_FromInteger(-50),
+                            field_10_anim.field_C_layer,
+                            field_10_anim.field_14_scale,
+                            field_19C_throwable_count,
+                            TRUE
+                        );
+                    }
+                }
+                field_FC_current_motion = eAbeStates::State_142_RockThrowStandingHold_429CE0;
+
+                if (gInfiniteGrenades_5076EC == 0)
+                {
+                    field_19C_throwable_count--;
+                }
+
+            }
+            else
+            {
+                field_FC_current_motion = eAbeStates::State_36_DunnoBegin_423260;
+            }
+        }
+        else
+        {
+            if (sInputObject_5009E8.isHeld(sInputKey_DoAction_4C65A4))
+            {
+                field_FC_current_motion = HandleDoAction_429A70();
+            }
+            else if (gBeeInstanceCount_5076B0 && word_5076AC)
+            {
+                field_FC_current_motion = eAbeStates::State_140_BeesStruggling_423F30;
+            }
+            else if (gnFrameCount_507670 - field_114_gnFrame > 150)
+            {
+                unsigned __int16 loaded = 0;
+                field_114_gnFrame = gnFrameCount_507670;
+                if (ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, 31, 0, 0))
+                {
+                    loaded |= 1;
+                }
+                if (ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, 32, 0, 0))
+                {
+                    loaded |= 2;
+                }
+                if (ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, 33, 0, 0))
+                {
+                    loaded |= 4;
+                }
+                short rnd = Math_RandomRange_450F20(0, 2);
```

Comment:
if you're gonna const the other stuff might as well const this as well, also this logic is OddBrain